api4/ldap.go

```diff
@@ -139,6 +139,8 @@ func linkLdapGroup(c *Context, w http.ResponseWriter, r *http.Request) {
 			newOrUpdatedGroup = group
 		} else {
 			group.DeleteAt = 0
+			group.DisplayName = ldapGroup.DisplayName
+			group.RemoteId = ldapGroup.RemoteId
```

Comment:
These values need to be updated when one restores a previously linked, then deleted, group. In fact `DisplayName` should have been updated here regardless of this feature change.